logstash-core/src/test/java/org/logstash/config/ir/compiler/OutputDelegatorTest.java

```diff
@@ -0,0 +1,247 @@
+        }
+    }
+
+    @Test
+    public void multiReceiveIncrementsEventCount() {
+        OutputDelegatorExt outputDelegator = constructOutputDelegator();
+        try {
+            outputDelegator.multiReceive(RUBY.getCurrentContext(), events);
+        } catch (InterruptedException e) {
+            fail("Multireceive error: " + e);
+        }
+
+        assertEquals(EVENT_COUNT, getMetricLongValue("in"));
+        assertEquals(EVENT_COUNT, getMetricLongValue("out"));
+    }
+
+    @Test
+    public void multiReceiveRecordsDurationInMillis() {
+        int delay = 100;
+        long millis = 0;
+        try {
+            fakeOutClass.setMultiReceiveDelay(delay);
+            OutputDelegatorExt outputDelegator = constructOutputDelegator();
+            try {
+                outputDelegator.multiReceive(RUBY.getCurrentContext(), events);
+            } catch (InterruptedException e) {
+                fail("Multireceive error: " + e);
+            }
+            millis = getMetricLongValue("duration_in_millis");
+        } finally {
+            fakeOutClass.setMultiReceiveDelay(0);
+        }
+
+        assertThat(millis).isCloseTo((long)delay, Percentage.withPercentage(10));
+    }
+
+    @Test
+    public void registersOutputPlugin() {
+        OutputDelegatorExt outputDelegator = constructOutputDelegator();
+        outputDelegator.register(RUBY.getCurrentContext());
+
+        assertEquals(1, fakeOutClass.getRegisterCallCount());
+    }
+
+    @Test
+    public void closesOutputPlugin() {
+        OutputDelegatorExt outputDelegator = constructOutputDelegator();
+        outputDelegator.doClose(RUBY.getCurrentContext());
+
+        assertEquals(1, fakeOutClass.getCloseCallCount());
+    }
+
+    @Test
+    public void singleConcurrencyStrategyIsDefault() {
+        OutputDelegatorExt outputDelegator = constructOutputDelegator();
+        IRubyObject concurrency = outputDelegator.concurrency(RUBY.getCurrentContext());
+        assertEquals(RUBY.newSymbol("single"), concurrency);
+    }
+
+    @Test
+    public void outputStrategyTests() {
+        RubySymbol[] outputStrategies = new RubySymbol[]{
+                RUBY.newSymbol("shared"),
+                RUBY.newSymbol("single"),
+                RUBY.newSymbol("legacy")
+        };
+
+        for (RubySymbol symbol : outputStrategies) {
+            fakeOutClass.setOutStrategy(RUBY.getCurrentContext(), symbol);
+            OutputDelegatorExt outputDelegator = constructOutputDelegator();
+
+            // test that output strategies are properly set
+            IRubyObject outStrategy = outputDelegator.concurrency(RUBY.getCurrentContext());
+            assertEquals(symbol, outStrategy);
+
+            // test that strategy classes are correctly instantiated
+            IRubyObject strategyClass = outputDelegator.strategy(RUBY.getCurrentContext());
+            assertThat(strategyClass instanceof OutputStrategyExt.AbstractOutputStrategyExt).isTrue();
```

Comment:
@danhermann I think this should check for the concrete output strategy class for the given symbol not the abstract one right?